pygmt/base_plotting.py

```diff
@@ -2127,18 +2128,10 @@ def velo(self, data=None, **kwargs):
 
         kind = data_kind(data)
 
-        extra_arrays = []
-        if "G" in kwargs and not isinstance(kwargs["G"], str):
-            if kind != "vectors":
-                raise GMTInvalidInput(
-                    "Can't use arrays for color if data is matrix or file."
-                )
-            extra_arrays.append(kwargs["G"])
-            del kwargs["G"]
 
-        if "S" in kwargs and not isinstance(kwargs["S"],str):
+        if "S" not in kwargs or ( "S" in kwargs and not isinstance(kwargs["S"],str)):
                 raise GMTInvalidInput(
-                    "Scaling is a required and has to be a string."
+                    "Scaling is a required argument and has to be a string."
                 )
 
```

Comment:
By adding the condition 
```python
if "S" not in kwargs
```
I was hoping triggering an error if `scaling` was not defined as an argument, for example by calling velo like that:
`figtest.velo(data=dataf)`

However, it didn't raise any error.  It seems that it doesn't run through this part of the code. Any idea why? 
Any suggestion on how to write a condition where I want `data`, `scaling`, `projection` and `region` defined as an argument?

Reply:
If no `data` is given, this will error out at the `kind = data_kind(data)` code line (as it requires `data` to be not `None`. Also, we don't need to check for `projection` or `region` (see https://github.com/GenericMappingTools/pygmt/pull/525#discussion_r527372247).

> However, it didn't raise any error. It seems that it doesn't run through this part of the code. Any idea why?

Not sure, the logic seems ok. Did you try just with `if "S" not in kwargs` (discarding the 'or' bit)?

Reply:
Thanks @weiji14 , it didnt do anything. It seems that if no additional argument is defined (`figtest.velo(data=dataf)`), this part of the script is not executed . I tried with `if 1==1` and nothing appears...